feat: add authentication and authorization using jwt(#40009)

permil-successive opened the pull request.

permil-successive committed:
feat: 🎸 add validation handler (middleware)

add validation handler (middleware) which validate input variables in
the request as per config file.

✅ Closes: #39524

permil-successive committed:
fix: 🐛 fix throw error on unwanted variable in required

permil-successive committed:
feat: 🎸 add auth middleware (Authentication & Authorization)

✅ Closes: #40009

permil-successive committed:
refactor: 💡 beautify the code

permil-successive committed:
Merge branch 'feature/39524' of https://github.com/permil-st/javascript-server into feature/40009

permil-successive committed:
refactor: 💡 add comments and break longer functions

add comments, break longer functions, merge import/export in single
statement wherever required

permil-successive committed:
Merge branch 'feature/39524' of https://github.com/permil-st/javascript-server into feature/40009

permil-successive committed:
fix: 🐛 remove extra supplied variable

permil-successive committed:
fix: 🐛 fix regex bug

permil-successive committed:
Merge branch 'feature/39524' of https://github.com/permil-st/javascript-server into feature/40009

Prabal555 commented on .env:
```diff
@@ -1,2 +1,3 @@
 PORT=9000
 ENV=dev
+SECRET_KEY=qwertyuiopasdfghjklzxcvbnm123456
```
.env should not be in .gitignore

Prabal555 commented on src/controllers/trainee/routes.ts:
```diff
@@ -1,13 +1,19 @@
 import Controller from './controller';
+import { validationHandler } from '../../libs';
+import validationConfig from './validations';
+import authorizationHandler from '../../libs/routes/authMiddleware';
```
not imported from the index file

Prabal555 commented on src/libs/routes/authMiddleware.ts:
```diff
@@ -0,0 +1,39 @@
+    code: '401',
+    message: 'Unathorised Access'
+  };
+
```
should be declared in the scope of use

permil-successive committed:
refactor: 💡 do changes as request in PR#10

remove unwanted route from server.ts, add a single response helper, fix
trainee delete route, fix custom validation function of trainee config,
fix response status code, use try/catch and pure function in
validationHandler.ts

✅ Closes: #39524

permil-successive committed:
fix: 🐛 fix bug in validation handler

for string datatype

permil-successive committed:
Merge branch 'feature/39524' of https://github.com/permil-st/javascript-server into feature/40009 and changes request in PR#11

Prabal555 commented on src/controllers/trainee/routes.ts:
```diff
@@ -1,13 +1,23 @@
+traineeRoutes.delete('/:id', authorizationHandler('trainee', 'delete'), validationHandler(validationConfig.delete), deleteFunction);
+=======
+traineeRoutes.delete('/:id', validationHandler(deleteConfig), deleteFunction);
+>>>>>>> 535d9e621d732678ec3a17ec90d6235829d88800
```
remove conflicts

permil-successive committed:
refactor: 💡 resolve merge conflicts
